### PR TITLE
Add in driver types and rejig code to make it easier to extend

### PR DESCRIPTION
**Pull request by BastiaanOlij**
Added support for https://github.com/godotengine/godot/pull/52562

And just rejigged code a bit so it becomes less work to add more to this.

### CODE REVIEW

File: include/godot_cpp/godot.hpp [hunk 1]
```diff
@@ -69,7 +65,7 @@ class GDExtensionBinding {
 		const GDNativeExtensionClassLibraryPtr library;
 		GDNativeInitialization *initialization;
 
-	public:
+	public:
```
Comment (vnen) [inline include/godot_cpp/godot.hpp]: Extra space here.
```suggestion
	public:
```

File: src/godot.cpp [hunk 2]
```diff
@@ -47,14 +47,15 @@ void *token = nullptr;
...
-GDExtensionBinding::Callback GDExtensionBinding::editor_terminate = nullptr;
+GDExtensionBinding::Callback GDExtensionBinding::init_callbacks[GDNATIVE_MAX_INITIALIZATION_LEVEL];
+GDExtensionBinding::Callback GDExtensionBinding::terminate_callbacks[GDNATIVE_MAX_INITIALIZATION_LEVEL];
+
+GDExtensionBinding::GDExtensionBinding() {
+	for (int i = 0; i < GDNATIVE_MAX_INITIALIZATION_LEVEL; i++) {
+		init_callbacks[i] = nullptr;
+		terminate_callbacks[i] = nullptr;
+	}
+}
```
Comment (vnen) [inline src/godot.cpp]: I don't think this constructor is ever called since this class is used statically.

Comment (BastiaanOlij) [reply src/godot.cpp]: Hmm, what would be a good way to clear out these buffers...

Comment (BastiaanOlij) [reply src/godot.cpp]: Owh I can move it into the constructor for InitObject..

**Comment by vnen** [comment]
Thanks!

**Pull request merged**